server/server.go

```diff
@@ -148,33 +176,60 @@ func (s *Server) getRoleMapping(IP string) (*mappings.RoleMappingResult, error)
 	return roleMapping, nil
 }
 
-// HealthResponse represents a response for the health check.
-type HealthResponse struct {
-	HostIP     string `json:"hostIP"`
-	InstanceID string `json:"instanceId"`
+func (s *Server) pollHealthcheck(interval time.Duration) {
+	for {
+		s.healthcheck()
+		time.Sleep(interval)
```

Comment:
Are you able to use a timer instead of `time.Sleep`? https://gobyexample.com/timers